src/Microsoft.Health.Dicom.Functions/Management/DurableClientProxy.cs

```diff
@@ -69,16 +80,20 @@ public static class DurableClientProxyFunctions
 
             if (string.IsNullOrWhiteSpace(instanceId))
             {
-                return new BadRequestResult();
+                return new HttpResponseMessage { StatusCode = HttpStatusCode.BadRequest };
```

Comment:
what is the reason for the change here?

Reply:
I think the majority of our controllers return an IActionResult? for consistency shouldn't we keep it like that?

Reply:
So ASP.NET Core relies on middleware to perform serialization. So typically APIs return `IActionResult` which typically encapsulate an HTTP Status Code and optionally some payload. Middleware then takes that payload and serializes it into the proper content type, like JSON or XML. However, Azure Functions are special. It seems they've custom-built a bunch of it including the DI service provider. So unfortunately while ASP.NET Core 3.1 uses the new `System.Text.Json` lib for serialization that is part of the BCL, Azure Functions relies on Json.NET (aka `Newtonsoft.Json`). Furthermore, you can seem to swap it out, so I am forced to serialize it myself and directly return `HttpResponseMessage` with the serialized payload.